DEVPROD-17803 Fix task group task drawdown logic

malikchaya2 opened the pull request:
DEVPROD-17803 
### Description
We want to not drawdown if the host is currently tearing down a task and it has not yet exceeded the max time.

ablack12 commented:
Can you add [DEVPROD-15777](https://jira.mongodb.org/browse/DEVPROD-15777) to the PR description and title?

ablack12 commented:
Also the title should explain what behavior is being fixed, rather than what the code is being changed (like, "fix task group task drawdown logic" or something)

ablack12 reviewed (approved):
LGTM with PR title improvement. Is it possible to add a test for this case also? https://github.com/evergreen-ci/evergreen/blob/2eac2da2f483ce2915f9bd273d49680250558d9c/units/host_drawdown_test.go#L173

malikchaya2 commented:
> Can you add [DEVPROD-15777](https://jira.mongodb.org/browse/DEVPROD-15777) to the PR description and title?

I should have explained more in the PR description. This is fixing something related I noticed while reading the code. I will put up a separate PR to address the concern in DEVPROD-15777.

malikchaya2 commented:
> LGTM with PR title improvement. Is it possible to add a test for this case also?
> 
> https://github.com/evergreen-ci/evergreen/blob/2eac2da2f483ce2915f9bd273d49680250558d9c/units/host_drawdown_test.go#L173

added a test and added another check to model/host/host.go that the test revealed was missing.
